src/components/graph/network-modification-tree-model.js

```diff
@@ -8,6 +8,16 @@
 import { getLayoutedNodes } from './layout';
 import { convertNodetoReactFlowModelNode } from './util/model-functions';
 
+// Function to count children nodes for a given parentId recursively in an array of nodes.
+function countNodes(nodes, parentId) {
+    return nodes.reduce((acc, n) => {
+        if (n.data.parentNodeUuid === parentId) {
+            acc += 1 + countNodes(nodes, n.id); // this node + its children
+        }
+        return acc;
+    }, 0);
+}
```

Comment:
sounds unoptimal to run "nodes.reduce" recursively on all nodes

there might be a better approach

Reply:
I added a comment for this